lib/ansible/modules/web_infrastructure/jboss.py

```diff
@@ -43,45 +43,140 @@
   - "Ensure no identically named application is deployed through the JBoss CLI"
+  - "At a minimum, url_password should be vaulted."
+  - "HTTP management API is supported in JBoss AS 7.1, Wildfly >= 8, and JBoss EAP >= 6."
+  - "JBoss 5 supports filesystem deployments only."
+  - "Filesystem deployments should be avoided in production environments where possible."
 author: "Jeroen Hoekx (@jhoekx)"
 """
 
 EXAMPLES = """
-# Deploy a hello world application
+# Deploy a hello world application using filesystem
 - jboss:
     src: /tmp/hello-1.0-SNAPSHOT.war
     deployment: hello.war
-    state: present
+    state: deployed
 
-# Update the hello world application
+# Update the hello world application using filesystem
 - jboss:
     src: /tmp/hello-1.1-SNAPSHOT.war
     deployment: hello.war
-    state: present
+    state: deployed
 
-# Undeploy the hello world application
+# Undeploy the hello world application using filesystem
 - jboss:
     deployment: hello.war
-    state: absent
+    state: undeployed
+
+# Deploy the hello world application using HTTP management API
+- jboss:
+    src: /tmp/hello-1.0-SNAPSHOT.war
+    deployment: hello.war
+    deployment_strategy: http
+    state: deployed
+    url_username: admin
+    url_password: admin
+
+# Update the hello world application using HTTP management API
+- jboss:
+    src: /tmp/hello-1.0-SNAPSHOT.war
+    deployment: hello.war
+    deployment_strategy: http
+    state: deployed
+    url_username: admin
+    url_password: admin
+
+# Undeploy the hello world application using HTTP management API
+- jboss:
+    deployment: hello.war
+    deployment_strategy: http
+    state: undeployed
+    url_username: admin
+    url_password: admin
+
 """
 
 import os
 import shutil
 import time
+import json
 from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
 
+def is_deployed(module):
 
-def is_deployed(deploy_path, deployment):
-    return os.path.exists(os.path.join(deploy_path, "%s.deployed" % deployment))
+    if module.params['deployment_strategy'] == 'http':
+        data_dict = {
+            "operation": "read-resource",
+            "address": {
+                "deployment": module.params['deployment']
+            },
+            "include-runtime": True
+        }
+
+        headers = {
+            'Content-Type': 'application/json'
+        }
+
+        data = json.dumps(data_dict)
+
+        resp, info = fetch_url(module, 'http://%s:9990/management' % module.params['hostname'], data=data, headers=headers)
+
+        return info['status'] == 200
+
+    elif module.params['deployment_strategy'] == 'filesystem':
+        return os.path.exists(os.path.join(module.params['deploy_path'], "%s.deployed" % module.params['deployment']))
+    else:
+        rc, stdout, stderr = module.run_command([module.params['cli_path'], '--connect', '--commands=deploy-info'])
```

Comment:
Thank you for prompt reply and for your effort on this.

While I agree it's cleaner to add `jboss_cli` as a separate module, the existing module is called plain `jboss` but it only handles (un)deployments. It could be confusing. On the other hand, I don't know how hard/painful is to rename a module. That's why I considered adding a CLI feature to `jboss` module.

Regarding `url_username`/`url_password`: In case of using those parameters for more than one (http) strategy `url_` part wouldn't be necessary. Maybe you assumed that, but just to be sure.